comm/p2p/libp2p.go

```diff
@@ -105,42 +114,70 @@ func (c Libp2pCommunication) UnSubscribe(
 
 /** Helper methods **/
 
+func (c Libp2pCommunication) StreamHandlerFunc(s network.Stream) {
+	c.ProcessMessagesFromStream(s)
+	err := s.Close()
+	if err != nil {
+		log.Warn().Msgf("Error closing incoming stream because of: %s", err.Error())
+	}
+}
+
+func (c Libp2pCommunication) ProcessMessagesFromStream(s network.Stream) {
+	remotePeerID := s.Conn().RemotePeer()
+	r := bufio.NewReader(s)
+	for {
+		msgBytes, err := ReadStream(r)
+		if err != nil {
+			return
+		}
+
+		var wrappedMsg comm.WrappedMessage
+		if err := json.Unmarshal(msgBytes, &wrappedMsg); nil != err {
+			log.Err(err).Msg("Error unmarshaling message")
+			return
+		}
+		wrappedMsg.From = remotePeerID
+
+		c.logger.Trace().Str(
+			"From", wrappedMsg.From.Pretty()).Str(
```

Comment:
This seems to be deprecated.

Reply:
Deprecated?

Reply:
// Pretty returns a base58-encoded string representation of the ID.
// Deprecated: use String() instead.